src/patterns/image-cdn/functions/forward-or-transform-image/forward-or-transform-image.handler.ts

```diff
@@ -0,0 +1,184 @@
+
+    console.debug(`Querying source image: ${sourceKey} in bucket ${bucketName}`)
+    const command = new GetObjectCommand({Bucket: bucketName, Key: sourceKey})
+    const imageObj = await s3.send(command)
+    const data = await imageObj.Body?.transformToByteArray()
+    if (!data) {
+      console.warn('Could not get object data', imageObj.Body)
+      return response
+    }
+    const resizedImage = await imageTransformer.transform(
+      Buffer.from(data),
+      {width, format, height},
+      {withoutEnlargement: true},
+    )
+
+    // TODO: set cache control on original image, and use it here
+    const cacheControl = 'max-age=31536000,public,immutable'
+    await s3.send(
+      new PutObjectCommand({
+        Body: resizedImage,
+        Bucket: bucketName,
+        ContentType: contentType,
+        Key: key,
+        StorageClass: 'STANDARD',
+        CacheControl: cacheControl,
+      }),
+    )
+
+    return {
+      ...response,
+      status: '200',
+      statusDescription: 'Found',
+      body: resizedImage.toString('base64'),
+      bodyEncoding: 'base64',
+      headers: {
+        ...response.headers,
+        'content-type': [{key: 'Content-Type', value: contentType}],
+        'cache-control': [{key: 'Cache-Control', value: cacheControl}],
+      },
+    }
+  } catch (error: unknown) {
+    const errorMessage = `Error while getting source image object "${sourceKey}": ${error}`
+    console.error('Error caught', error)
+
+    return {
+      ...response,
+      status: '404',
+      statusDescription: 'Not Found',
+      body: errorMessage,
+      bodyEncoding: 'text',
+      headers: {
+        ...response.headers,
+        'content-type': [{key: 'Content-Type', value: 'text/plain'}],
+      },
+    }
+  }
+}
+
+function getOriginBucketName(event: CloudFrontResponseEvent): string | undefined {
+  const {
+    cf: {
+      request: {origin},
+    },
+  } = event.Records[0]
+  if (origin?.s3) {
+    const domainName = origin.s3.domainName
+    return parseS3BucketName(domainName)
+  } else if (origin?.custom) {
+    const domainName = origin.custom.domainName
+    if (!domainName) {
+      console.warn('could not find custom origin domain name, returning ...')
+      return undefined
+    } else if (!domainName.includes('s3-website')) {
+      console.warn('custom origin is not an s3-website (unsupported), returning ...')
+      return undefined
+    } else {
+      return parseS3BucketName(domainName)
+    }
+  } else {
+    return undefined
+  }
+}
+
+export function parseS3BucketName(domainName: string): string | undefined {
+  const bucketNameRegex = /[^/]+(?=.(?:s3|s3-website)\.(?:[a-z0-9-]+)\.amazonaws\.com)/i
+  const matches = domainName.match(bucketNameRegex)
+  if (matches?.length) {
+    return matches[0]
+  } else {
+    return undefined
+  }
+}
+
+function parseFormat(fmt?: unknown): Format | undefined {
+  if (!isString(fmt)) {
+    return undefined
+  }
+  switch (fmt) {
+    case 'jpeg':
```

Comment:
I know this isn't new code, but just checking, any chance we'd ever get 'jpg'? Should we cater for that?

Reply:
We normalize the jpg extension in the function that parses the URI and query params, so it should never happen !

https://github.com/buildigo/aws-cdk-patterns/pull/31/files#diff-c6501f43211c54857f7642099e92341f7264c3ebe797370f290b1c979027d0fcR48-R51